botorch/models/higher_order_gp.py

```diff
@@ -361,39 +362,60 @@ def __init__(
         covar_modules: Optional[List[Kernel]] = None,
         num_latent_dims: Optional[List[int]] = None,
         learn_latent_pars: bool = True,
-        first_dim_is_batch: bool = False,
         latent_init: str = "default",
         outcome_transform: Optional[OutcomeTransform] = None,
         input_transform: Optional[InputTransform] = None,
     ):
         r"""A HigherOrderGP model for high-dim output regression.
 
         Args:
-            train_X: Training inputs
-            train_Y: Training targets
+            train_X: A `batch_shape x n x d`-dim tensor of training inputs.
+            train_Y: A `batch_shape x n x output_shape`-dim tensor of training targets.
             likelihood: Gaussian likelihood for the model.
             covar_modules: List of kernels for each output structure.
             num_latent_dims: Sizes for the latent dimensions.
             learn_latent_pars: If true, learn the latent parameters.
-            first_dim_is_batch: If true, the first dimension of train_Y should be
-                regarded as a batch dimension (e.g. predicting batches of tensors).
             latent_init: [default or gp] how to initialize the latent parameters.
         """
 
         if input_transform is not None:
             input_transform.to(train_X)
 
+        # infer the dimension of `output_shape`.
+        num_output_dims = train_Y.dim() - train_X.dim() + 1
+        if num_output_dims not in [train_Y.dim() - 1, train_Y.dim() - 2]:
+            raise NotImplementedError(
+                "HigherOrderGP currently only supports 1-dim `batch_shape`."
+            )
```

Comment:
why not just check that `len(train_X.shape[:-2])` is at most of length 1? seems less convoluted